### PR TITLE
Make `SingleQubitCompare` output registers directional

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes https://github.com/quantumlib/Qualtran/issues/388

cc @fdmalone

### CODE REVIEW

File: cirq-ft/cirq_ft/algos/arithmetic_gates.py [hunk 1]
```diff
@@ -222,7 +222,15 @@ class SingleQubitCompare(infra.GateWithRegisters):
 
     @cached_property
     def signature(self) -> infra.Signature:
-        return infra.Signature.build(a=1, b=1, less_than=1, greater_than=1)
```
Comment (fdmalone) [inline cirq-ft/cirq_ft/algos/arithmetic_gates.py:225]: Should BiQubitsMixer be updated too?

**Review by fdmalone — APPROVED** [review]

File: cirq-ft/cirq_ft/algos/arithmetic_gates.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-ft/cirq_ft/algos/arithmetic_gates.py:225]: I think so, but I'd recommend to open a separate issue in Qualtran. I haven't looked at the decomposition too closely, but I'm not sure if `ancilla` doesn't register doesn't need to be part of the signature at all or whether it should be a directional register like the target of an `And` gate. 

Maybe @NoureldinYosri can clarify this on the separate issue and then we can send a PR to update it and it's usage. Also, it looks like we are not really using `BiQubitsMixer(adjoint=True)` anywhere.

Comment (fdmalone) [reply cirq-ft/cirq_ft/algos/arithmetic_gates.py:225]: Ok. I'll bump by cirq version locally and see. This is good to merge.

Comment (fdmalone) [reply cirq-ft/cirq_ft/algos/arithmetic_gates.py:225]: the Adjoint=True is used during the decomposition I believe.

Screenshot 2023-10-09 at 4 13 04 PM

Comment (NoureldinYosri) [reply cirq-ft/cirq_ft/algos/arithmetic_gates.py:225]: @tanujkhattar @fdmalone it should be directional like `And`

**Pull request merged**